CollectiveFocus/CFM_Backend
# [CFM_100] Update template.yaml to deploy backend

Ticket: https://trello.com/c/jps5SRwX/108-cfm100-deploy-backend

### Updates

1. [CommunityFridgeMapApi/functions/dev/get_all](https://github.com/CollectiveFocus/CFM_Backend/compare/CFM_100_JP?expand=1#diff-1e5a4421cc3f6d6996824bec189c1746ae0318a206d67b00e32698cad8c67d21)
    * Delete GetAllFunction function and files dependent on it
2. [CommunityFridgeMapApi/template.yaml](https://github.com/CollectiveFocus/CFM_Backend/compare/CFM_100_JP?expand=1#diff-499907e35312d0a5f51204c2474c5b1a1cbe7dc127d59c3ef751559f10af07db)
    * Dynamodb Resource requires to only define the HASH and RANGE keys so I had to delete all the other fields
3. [README.md](https://github.com/CollectiveFocus/CFM_Backend/compare/CFM_100_JP?expand=1#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)
    * Added Database Table Design document to the README
    
### Testing

### Unit Tests
1. Run Unit Tests: `coverage run -m pytest tests/unit -v`

### QA
   * build container: `sam build --use-container`
   * Run dynamodb locally and create tables. Follow README instructions to do so if you haven't already https://github.com/CollectiveFocus/CFM_Backend/blob/dev/README.md#setup-local-database-connection
   * Run any of the APIs


Deployed Endpoints:   
Confirm that the following endpoints work:
https://xsf433l0bl.execute-api.us-east-1.amazonaws.com/Prod/v1/fridges/2fish5loavesfridge
https://xsf433l0bl.execute-api.us-east-1.amazonaws.com/Prod/v1/fridges
